PocketKit/Sources/PocketKit/Item/ReadableHostViewController.swift

```diff
@@ -99,7 +91,40 @@ class ReadableHostViewController: UIViewController {
 
     @objc
     private func archiveArticle() {
-        readableViewModel.archiveArticle()
+        readableViewModel.archive()
+    }
+
+    @objc
+    private func unarchiveArticle() {
+        readableViewModel.unarchive { [weak self] success in
+            if success, let archiveButton = self?.getArchiveButton() {
+                self?.navigationItem.rightBarButtonItems?[2] = archiveButton
+            }
+        }
```

Comment:
Should I add a check for the `rightBarButtonItems.count` > 3, but we know this already when initializing it?

Reply:
Boundary checks would be good here. It works now but is brittle and may break on change.
Maybe an Enum with each buttons index, to avoid crossover?